lib/bridge/MatrixHandler.js

```diff
@@ -770,6 +769,68 @@ MatrixHandler.prototype._onMessage = Promise.coroutine(function*(req, event) {
     yield Promise.all(promises);
 });
 
+MatrixHandler.prototype._sendIrcAction = Promise.coroutine(
+    function*(req, ircRoom, ircUser, ircAction, event) {
+    let self = this;
+
+    if (event.content.msgtype !== "m.text" ||
+        !(ircUser.unsafeClient && ircUser.unsafeClient.wouldSend)) {
+        yield self.ircBridge.sendIrcAction(ircRoom, ircUser, ircAction);
+    }
+    else {
+        let text = event.content.body;
+        let wouldSend = ircUser.unsafeClient.wouldSend(ircRoom.channel, text);
+        let lineLimit = 2;
+        if (wouldSend.length > lineLimit) {
+            // Message body too long - send a file to HS instead and send link to IRC
+
+            try {
+                // Upload as a file and get URI
+
+                let fileName = Date().replace(/\s/g, '_') + ".txt";
+                let response = yield self.ircBridge.sendMatrixPlaintextFile(fileName, text);
+                let result = JSON.parse(response);
+                event.content.url = result.content_uri;
+
+                // Send the link to IRC
+                event.content.msgtype = "m.file";
+                event.content.body = fileName;
```

Comment:
Your comment is unhelpful, as you actually do the sending on `:805`. You don't explain why you're clobbering fields on the event though. You should say why you're doing that.